plan/eliminate_projection.go

```diff
@@ -158,9 +157,6 @@ func (pe *projectionEliminater) eliminate(p LogicalPlan, replace map[string]*exp
 	if !(isProj && canEliminate && canProjectionBeEliminatedLoose(proj)) {
 		return p
 	}
-	if join, ok := p.Parents()[0].(*LogicalJoin); ok {
```

Comment:
Why we could remove this?

Reply:
@shenli projection elimination is done before aggregation push down, we can remove it safely.